scripts/fast-reboot-filter-routes.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+import json
+import sys
+import os
+import utilities_common.cli as clicommon
+import syslog
+import traceback
+import click
+from builtins import str #for unicode conversion in python2
+from swsscommon.swsscommon import ConfigDBConnector
+
+ROUTE_IDX = 1
+
+def get_connected_routes():
+    cmd = 'sudo vtysh -c "show ip route connected json"'
+    connected_routes = []
+    try:
+        output, ret = clicommon.run_command(cmd, return_cmd=True)
+        if ret != 0:
+            click.echo(output.rstrip('\n'))
+            sys.exit(ret)
+        if output is not None:
+            route_info = json.loads(output)
+            for route, info in route_info.items():
+                connected_routes.append(route)
+    except Exception:
+        ctx = click.get_current_context()
+        ctx.fail("Unable to get connected routes from bgp")
```

Comment:
What in intended behavior when we hit this exception? If this returns non-zero code to bash, fast-reboot script will still continue as the call to filer-routes is after `set +e` in bash script. Do you still want the fast-reboot to proceed when routes are not filtered?

Reply:
Fast-reboot requires BGP is enabled, this will return non-zero code only in case BGP is not used and hence is not relevant for fast-reboot if I understand correctly.
However this won't fast-reboot to fail but it might make it exceed the dataplane downtime if filtering failed and switch has rather large amount of routes stored.

Reply:
I will leave it to you. To me it sounds like this will impact dataplane downtime as the PR is fixing that when the connected routes aren't preserved. So if you wish you could set this as a required criteria - by moving the call the preserve before set +e. Just FYI.